File: dsa/algorithms/sorting_algos/insertion_sort/insertion_sort.java
package insertion_sort;

public class insertion_sort {
    public static void main(String[] args) {
        int[] list = {3, 2, 3, 1, 2, 4, 3, 4, 3, 2};
        algo(list);
        
    }

    static void algo(int[] list) {
        for (int i = 1; i < list.length; i++) {
            int j = i;
            while (j > 0 && list[j - 1] > list[j]) {
                int temp = list[j];
                list[j] = list[j-1];
                list[j-1] = temp;
                j--;
            }
        }
        for (int i = 0; i < list.length; i++) {
            System.out.printf("%s ", list[i]);
        }
        System.out.println();
    }
}

File: dsa/algorithms/sorting_algos/insertion_sort/insertion_sort.py
def insertion_sort(arr):
    for i in range(1, len(arr)):
        j = i
        while arr[j - 1] > arr[j] and j > 0:
            arr[j-1], arr[j] = arr[j], arr[j-1]
            j-=1

    return arr

print(insertion_sort([12, 3, 2, 1, 2, 3, 4, 2, 1, 111, 222, 111, 22, 1.2]))